app/models/jobstatusdata.rb

```diff
@@ -32,10 +32,15 @@ def initialize(info, cluster=OODClusters.first, extended=false)
     self.cluster = cluster.id.to_s
     self.cluster_title = cluster.metadata.title ||  cluster.id.to_s.titleize
     self.walltime_used = info.wallclock_time.to_i > 0 ? pretty_time(info.wallclock_time) : ''
+    self.walltime = pretty_time(info.wallclock_limit)
     self.queue = info.queue_name
+    self.submission_time= info.submission_time
     if info.status == :running || info.status == :completed
       self.nodes = node_array(info.allocated_nodes).reject(&:blank?)
       self.starttime = info.dispatch_time.to_i
+      self.queued_time = starttime > 0 ? pretty_time(self.starttime- self.submission_time.to_i) : '';
+    else
+      self.queued_time = pretty_time(Time.now().to_i - self.submission_time.to_i)
```

Comment:
The problem with this logic is that there are other statues (suspended, hold, etc.) So I guess this "time in queue" value could be "off" by a bit, if it switches from queue to hold or queue to suspend to queue etc. It seems like a disclaimer would be nice... not sure how we would do that. The disclaimer would be something like: (This value was calculated by current time or start time from submission time. If the job entered into a suspended or held or other non-queued state during this time this value could be incorrect.). Maybe a * next to the key and a popover would display this message. Or maybe the disclaimer is in the text field after the value, italicized and greyed out.

Also maybe we should have the label be "Time in Queue" or "Duration in Queue" something to indicate we are talking about duration and not the timestamp when the state changed to "queued".